plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/AltibaseDataSource.java

```diff
@@ -0,0 +1,656 @@
+    }
+
+    /**
+     * Returns a specific Altibase user. 
+     */
+    @Association
+    public AltibaseUser getUser(DBRProgressMonitor monitor, String name) throws DBException {
+        return userCache.getObject(monitor, this, name);
+    }
+    
+    /**
+     * Altibase Roles
+     */
+    static class RoleCache extends JDBCObjectCache<AltibaseDataSource, AltibaseRole> {
+        @NotNull
+        @Override
+        protected JDBCStatement prepareObjectsStatement(@NotNull JDBCSession session, 
+                @NotNull AltibaseDataSource owner) throws SQLException {
+            return session.prepareStatement(
+                "SELECT * "
+                + " FROM SYSTEM_.SYS_USERS_ u "
+                + " WHERE u.user_type = 'R' AND u.user_name <> 'PUBLIC'"
+                + " ORDER BY user_name");
+        }
+
+        @Override
+        protected AltibaseRole fetchObject(@NotNull JDBCSession session, @NotNull AltibaseDataSource owner, 
+                @NotNull JDBCResultSet resultSet) throws SQLException, DBException {
+            return new AltibaseRole(owner, resultSet);
+        }
+    }
+    
+    /**
+     * Returns Altibase roles 
+     */
+    @Association
+    public Collection<AltibaseRole> getRoles(DBRProgressMonitor monitor) throws DBException {
+        return roleCache.getAllObjects(monitor, this);
+    }
+
+    /**
+     * Returns a specific Altibase role.
+     */
+    @Association
+    public AltibaseRole getRole(DBRProgressMonitor monitor, String name) throws DBException {
+        return roleCache.getObject(monitor, this, name);
+    }
+    
+    /**
+     * Returns Altibase grantee. 
+     */
+    public AltibaseGrantee getGrantee(DBRProgressMonitor monitor, String name) throws DBException {
+        AltibaseUser user = userCache.getObject(monitor, this, name);
+        if (user != null) {
+            return user;
+        }
+        return roleCache.getObject(monitor, this, name);
+    }
+    
+    
+    ///////////////////////////////////////////////
+    // Replications
+    static class ReplicationCache extends JDBCStructLookupCache<GenericStructContainer, AltibaseReplication, AltibaseReplicationItem> {
+        
+        final AltibaseDataSource dataSource;
+        
+        protected ReplicationCache(AltibaseDataSource dataSource) {
+            super("Replication");
+            this.dataSource = dataSource;
+            setListOrderComparator(DBUtils.<AltibaseReplication>nameComparatorIgnoreCase());
+        }
+        
+        public AltibaseDataSource getDataSource() {
+            return dataSource;
+        }
+        
+        @NotNull
+        @Override
+        public JDBCStatement prepareLookupStatement(@NotNull JDBCSession session, @NotNull GenericStructContainer owner, 
+                @Nullable AltibaseReplication object, @Nullable String objectName) throws SQLException {
+            String replName = (object != null) ? object.getName() : "";
```

Comment:
```suggestion
            String replName = (object != null) ? object.getName() : objectName;
```

Reply:
objectName can be returned and used here, too.

Reply:
Okay, it will be more readable.